src/cppHeaderTransfer.ts

```diff
@@ -43,34 +43,25 @@ export class CppHeaderTransfer extends assemblyscript.ASTBuilder {
     }
 
     const cppTypeString: string | undefined = this.typeMap.get(typeString);
-    if (cppTypeString) {
-      return cppTypeString;
-    } else {
-      return typeString;
-    }
+    return cppTypeString ?? typeString;
   }
 
   public visitEnumDeclaration(node: assemblyscript.EnumDeclaration, isDefault?: boolean): void {
-    // console.log(node.range.source.text.substring(node.range.start, node.range.end));
     this.writeContentWithBreakLine(`enum class ${node.name.text}{`);
     for (const item of node.values) {
-      if (
-        item["initializer"] &&
-        item.initializer.range &&
-        item.initializer.range.end - item.initializer.range.start > 0
-      ) {
+      if (item.initializer === null) {
+        this.writeContentWithBreakLine(`${item.name.text},`);
+      } else {
+        const expr = item.initializer;
         if (
-          this.genValueFromRange(item.initializer.range).startsWith("0x") ||
-          this.genValueFromRange(item.initializer.range).startsWith("0X")
+          expr instanceof assemblyscript.IntegerLiteralExpression ||
+          (expr instanceof assemblyscript.UnaryPrefixExpression &&
+            expr.operand instanceof assemblyscript.IntegerLiteralExpression)
         ) {
-          this.writeContentWithBreakLine(
-            `${item.name.text}=0x${Number(item.initializer["value"]["low"]).toString(16)},`
-          );
+          this.writeContentWithBreakLine(`${item.name.text}=${this.genValueFromRange(expr.range)},`);
         } else {
-          this.writeContentWithBreakLine(`${item.name.text}=${Number(item.initializer["value"]["low"]).toString()},`);
+          throw new TypeError(`enum expr not supported: ${item.name.text}=${this.genValueFromRange(expr.range)}.`);
```

Comment:
provide supported expression type information (unary expression with integer literal and integer literal)